src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DepartureEntity.kt

```diff
@@ -12,7 +13,11 @@ import javax.persistence.ManyToOne
 import javax.persistence.Table
 
 @Repository
-interface DepartureRepository : JpaRepository<DepartureEntity, UUID>
+interface DepartureRepository : JpaRepository<DepartureEntity, UUID> {
+
+  @Query(nativeQuery = true)
+  fun findAllDeparturesByBookingId(bookingId: UUID? = null): List<DepartureEntity>
```

Comment:
I don't think this method is necessary as you can just use `bookingEntity.departures` to get this information.

Reply:
I thought so initially, but for some reason its lose the value we for `departures` in the same method.  Hence i couldnt use bookingEntity.departures.

Reply:
I've had a similar problem before - just calling `departureRepository.save` isn't enough to update the booking with the new departure - you also need to attach the departure to the booking using something like `booking.departures += departureEntity`.

Reply:
it worked, thank